### DISCUSSION TITLE
Macros & alias for multiple cmd2 commands

### DISCUSSION THREAD

**Discussion by Sripadvallabh** [python-cmd2/cmd2#1320] [General]
Hi,

Is there a way to add multiple cmd2 commands into an alias or macro in any way?

Bash supports the '&&' or ';' operator to combine multiple commands; is there a way to do the same in cmd2???

**Comment by kmvanbrunt** [comment]
`cmd2` does not support  '&&' or ';' operators.

However, you can run multiple commands in scripts.
https://cmd2.readthedocs.io/en/stable/features/scripting.html

For a basic command script, create a file with one command per line.
```
command1 arg arg
command2 arg arg
command3 arg arg
```

Run it using `run_script script.txt`.

If you want, you can create an alias to run the script.
https://cmd2.readthedocs.io/en/stable/features/shortcuts_aliases_macros.html#aliases

`alias create my_alias run_script script.txt`

Lastly, you can use a startup script to recreate your aliases each time you start the application.
https://cmd2.readthedocs.io/en/stable/features/startup_commands.html?#startup-script

**Reply by Sripadvallabh** [reply]
thanks kevin for your answer.

The run_script looks great; My only concern is it clutters my workspace by having too many files around for each of those aliases i want to use. If there is a way to add multiple commands in one alias inside the startup script itself it would be great.

Please let me know if its possible.

**Reply by kmvanbrunt** [reply]
There isn't a way to run multiple commands without a script.
I typically place my scripts in their own directory so they don't clutter my application directory.
